ecosystem/typescript/sdk_v2/src/bcs/move-types/primitives.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { AnyNumber } from "../../types";
+import { Deserializer } from "../deserializer";
+import { Serializable, Serializer } from "../serializer";
+
+export class Bool extends Serializable {
+  public value: boolean;
+
+  constructor(value: boolean) {
+    super();
+    this.value = value;
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeBool(this.value);
+  }
+
+  deserialize(deserializer: Deserializer) {
+    this.value = deserializer.deserializeBool();
+  }
+
+  static deserialize(deserializer: Deserializer): Bool {
+    return new Bool(deserializer.deserializeBool());
+  }
+}
+export class U8 extends Serializable {
+  public value: number;
```

Comment:
Curious, what happen if the user create an U8 with a number > than u8 can hold? i.e: values that are > 255. Should we do validation on u8 to u256?

Reply:
oh yes, good point! We already have a decorator for that, I will add those

Reply:
okay so I tried a bunch of ways- but it messes with the underlynig classes/type a ton, so I'm not going to do it.

`serializeX` and `deserializeX` already run the checks- so the only time you'll not see an error is if you create a `U8` that has an invalid number but never serialize it. If they try to deserialize a value that's too large into a `U8` it will also error. so I feel like we don't need to care about the case where they do `new U8(1500)`

I can add checks for it, but it mostly just added more problems than it was worth...